### PR TITLE
[0] BaseComponents for screens and components containing Stores and Products

### PR DESCRIPTION
**Pull request by tommypoa**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
### General Changes
- All Text elements changed to that of `BaseComponents`

### Deletions / Replacements
- `Button` in exchange for `ButtonContainer` in most screens – the major deviation from what @thumn did in PR (see Relevant Links).
- `StoreDetailsScreen.js` deleted, as per Figma
- Product.js replaced with ProductCard.js, for alignment with our broader codebase

### Online sources
- NIL

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
- #28: @thumn PR on BaseComponents

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
- General Changes -- go through all files changed to see if the BaseComponents I used to replace are correct, according to the Figma
- Test on App – go through all screens and interactions with Stores and Products. Namely:
   - MapScreen (click on the products on the `BottomSheet`
   - ProductDetailedScreen
   - StoreListScreen (verify stores are still rendering correctly)

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
- Realised that `MaterialCommunityIcons` is used in `StoreCard.js` – to phase out and use `FontAwesome` instead
## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"
- All behaviour on `MapScreen.js` and `StoreListScreen.js` working as before.
### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
- NIL - no major styling changes.

CC: @wangannie @12aschen 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/product/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        navigation.navigate('ProductDetails', {
+          currentProduct: product,
+          store
+        })
+      }>
+      <Image
+        source={require('../../assets/images/robot-dev.png')}
+        style={{ width: 50, height: 50, borderRadius: 80 / 2 }}
+      />
+      <Caption>{product.name}</Caption>
```
Comment (wangannie) [inline components/product/ProductCard.js]: Product name is Body not Caption.

File: components/product/SpecialNotice.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
-import { Notice } from '../../styled/product';
+import { Body } from '../BaseComponents';
 
 function SpecialNotice() {
   return (
-    <Notice>
+    <Body>
```
Comment (wangannie) [inline components/product/SpecialNotice.js]: This is a Caption, not Body

Comment (wangannie) [reply components/product/SpecialNotice.js]: Color is secondaryText

Comment (tommypoa) [reply components/product/SpecialNotice.js]: SpecialNotice deleted and moved to ProductInfo.js in #34. Will implement changes as part of that PR!
